lib/child_process.js

```diff
@@ -19,15 +19,20 @@ const ChildProcess = exports.ChildProcess = child_process.ChildProcess;
 exports.fork = function(modulePath /*, args, options*/) {
 
   // Get options and args arguments.
-  var options, args, execArgv;
-  if (Array.isArray(arguments[1])) {
-    args = arguments[1];
-    options = util._extend({}, arguments[2]);
-  } else if (arguments[1] && typeof arguments[1] !== 'object') {
-    throw new TypeError('Incorrect value of args option');
-  } else {
-    args = [];
-    options = util._extend({}, arguments[1]);
+  var execArgv;
+  var options = {};
+  var args = [];
+  var pos = 1;
+  if (Array.isArray(arguments[pos])) {
```

Comment:
> The code I'm replacing accessed arguments[1] without any bounds check, so it was already not being optimized.

Yes, that's quite possibly the case.

> This isn't going to be something that's benchmark-able because each fork() call is going to spin up a completely separate Node.js instance

That's right but it would be good for consistency.  Elsewhere in lib/child_process.js we're quite diligent about checking `arguments.length`.  Also, it makes `--trace_deopt` less noisy, which is never a bad thing.